src/std/5-std-maps.md

```diff
@@ -0,0 +1,177 @@
+# Maps
+
+The **map** module defines the `map` type and all the vector manipulation related functions.
+
+A `map` is an associative container that can store heterogeneous types.
+
+All the given examples assume the following imports:
+```
+let map = import('std/map');
+let io = import('std/io');
+```
+
+## Functions
+ - [new](#new)
+
+## `map` Member Functions
+ - [len](#len)
+ - [insert](#insert)
+ - [erase](#erase)
+ - [get](#get)
+ - [operator []](#operator-)
+ - [find](#find)
+ - [each](#each)
+ - [next](#next)
+ 
+### new
+```
+map.new(key1, value1, key2, value2, ...) -> map
+```
+Creates a new map using the given key/value pairs. The keys must be convertible to `string`s. Also, it must correspond a value to each key, meaning that the number of arguments must always be even
+
+Example:
+```
+let m = map.new(0, "Zero", 1, "One", "Answer?", 42);
+io.println(m);
+```
+
+Gives the output:
+```
+{Answer?: 42, 0: Zero, 1: One}
+```
+ 
+Note that the elements are not stored in the order they are given. They are lexicographically sorted
```

Comment:
They are not sorted at all - c++'s `unordered_map`.
From cppreference's `unordered_map` page:
> Internally, the elements are not sorted in any particular order, but organized into buckets. Which bucket an element is placed into depends entirely on the hash of its key. This allows fast access to individual elements, since once the hash is computed, it refers to the exact bucket the element is placed into.

Reply:
Oops, I was probably thinking of `std::map` when I wrote this